app/gui/src/dashboard/services/ProjectManager.ts

```diff
@@ -503,6 +487,10 @@ export default class ProjectManager {
 
   /** Delete a project. */
   async deleteProject(params: Omit<DeleteProjectParams, 'projectsDirectory'>): Promise<void> {
+    const cached = this.internalProjects.get(params.projectId)
```

Comment:
This should be avoided in the future - cache management is purely the tanstack query responsibility.

Reply:
honestly yeah. but i think this is more or less a workaround for issues in the engine API, the alternative is we'd have to implement localbackend through tanstack query instead somehow.

specifically: the "cache" here is so that we actually remember the state of the projects because (iirc?) there is no way to actually get the state of the projects though the Project Manager... if we could then none of this would be necessary

Reply:
But we have the same cache in the tanstack query, haven't we?